diff --git a/seeed-voicecard.c b/seeed-voicecard.c
index f9df09c..33d1c95 100644
--- a/seeed-voicecard.c
+++ b/seeed-voicecard.c
@@ -100,7 +100,7 @@ static int seeed_voice_card_startup(struct snd_pcm_substream *substream)
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
 	struct seeed_card_data *priv =	snd_soc_card_get_drvdata(rtd->card);
 	struct seeed_dai_props *dai_props =
-		seeed_priv_to_props(priv, rtd->id);
+		seeed_priv_to_props(priv, rtd->dai_link->id);
 	int ret;
 
 	ret = clk_prepare_enable(dai_props->cpu_dai.clk);
@@ -130,7 +130,7 @@ static void seeed_voice_card_shutdown(struct snd_pcm_substream *substream)
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
 	struct seeed_card_data *priv =	snd_soc_card_get_drvdata(rtd->card);
 	struct seeed_dai_props *dai_props =
-		seeed_priv_to_props(priv, rtd->id);
+		seeed_priv_to_props(priv, rtd->dai_link->id);
 
 	snd_soc_rtd_to_cpu(rtd, 0)->driver->playback.channels_min = priv->channels_playback_default;
 	snd_soc_rtd_to_cpu(rtd, 0)->driver->playback.channels_max = priv->channels_playback_default;
@@ -150,7 +150,7 @@ static int seeed_voice_card_hw_params(struct snd_pcm_substream *substream,
 	struct snd_soc_dai *cpu_dai = snd_soc_rtd_to_cpu(rtd, 0);
 	struct seeed_card_data *priv = snd_soc_card_get_drvdata(rtd->card);
 	struct seeed_dai_props *dai_props =
-		seeed_priv_to_props(priv, rtd->id);
+		seeed_priv_to_props(priv, rtd->dai_link->id);
 	unsigned int mclk, mclk_fs = 0;
 	int ret = 0;
 
@@ -415,7 +415,7 @@ static int seeed_voice_card_dai_init(struct snd_soc_pcm_runtime *rtd)
 	struct snd_soc_dai *codec = snd_soc_rtd_to_codec(rtd, 0);
 	struct snd_soc_dai *cpu = snd_soc_rtd_to_cpu(rtd, 0);
 	struct seeed_dai_props *dai_props =
-		seeed_priv_to_props(priv, rtd->id);
+		seeed_priv_to_props(priv, rtd->dai_link->id);
 	int ret;
 
 	ret = simple_util_init_dai(codec, &dai_props->codec_dai);
